tests/test_dataloader.py

```diff
@@ -394,6 +394,87 @@ def test_finetuning_dataloader_small_data(dataset_size: int,
         shutil.rmtree(tiny_dataset_folder_path)
 
 
+@pytest.mark.parametrize('split', ['train', 'custom', 'data'])
+def test_finetuning_dataloader_custom_split(tmp_path: pathlib.Path, split: str):
+    tokenizer_name = 'gpt2'
+    max_seq_len = 2048
+    tiny_dataset_folder_path = str(tmp_path)
+    tiny_dataset_path = os.path.join(tiny_dataset_folder_path, 'data',
+                                     f'{split}-00000-of-00001.jsonl')
+    if dist.get_global_rank() == 0:
+        make_tiny_ft_dataset(path=tiny_dataset_path, size=16)
+
+    cfg = {
+        'name': 'finetuning',
+        'dataset': {
+            'hf_name': tiny_dataset_folder_path,
+            'split': split,
+            'max_seq_len': max_seq_len,
+            'decoder_only_format': True,
+            'allow_pad_trimming': False,
+            'packing_ratio': None,
+            'shuffle': True,
+        },
+        'drop_last': False,
+        'num_workers': 4,
+        'pin_memory': False,
+        'prefetch_factor': 2,
+        'persistent_workers': False,
+        'timeout': 0
+    }
+
+    cfg = om.create(cfg)
+
+    tokenizer = build_tokenizer(
+        tokenizer_name=tokenizer_name,
+        tokenizer_kwargs={'model_max_length': max_seq_len},
+    )
+
+    _ = build_finetuning_dataloader(cfg, tokenizer, 4)
+
+
+def mock_get_file(path: str, destination: str, overwrite: bool = False):
+    make_tiny_ft_dataset(path=destination, size=16)
+
+
+@pytest.mark.parametrize('split', ['train', 'custom', 'data'])
+def test_finetuning_dataloader_custom_split_remote(tmp_path: pathlib.Path,
+                                                   split: str, monkeypatch: pytest.MonkeyPatch):
+    tokenizer_name = 'gpt2'
+    max_seq_len = 2048
+
+    cfg = {
+        'name': 'finetuning',
+        'dataset': {
+            'hf_name': 's3://test-bucket/path/to/data',
+            'split': split,
+            'max_seq_len': max_seq_len,
+            'decoder_only_format': True,
+            'allow_pad_trimming': False,
+            'packing_ratio': None,
+            'shuffle': True,
+        },
+        'drop_last': False,
+        'num_workers': 4,
+        'pin_memory': False,
+        'prefetch_factor': 2,
+        'persistent_workers': False,
+        'timeout': 0
+    }
+
+    cfg = om.create(cfg)
+
+    tokenizer = build_tokenizer(
+        tokenizer_name=tokenizer_name,
+        tokenizer_kwargs={'model_max_length': max_seq_len},
+    )
+
+    with monkeypatch.context() as m:
```

Comment:
nit: I think instead you can just add this to the top of your function:
```
@patch('llmfoundry.data.finetuning.dataloader.get_file',
       new=mock_get_file)
```
and you don't need to have an extra function parameter also